pilot/pkg/xds/debug.go

```diff
@@ -987,10 +988,31 @@ func (s *DiscoveryServer) forceDisconnect(w http.ResponseWriter, req *http.Reque
 	_, _ = w.Write([]byte("OK"))
 }
 
+func cloneProxy(proxy *model.Proxy) *model.Proxy {
```

Comment:
Interesting! I didn't think of this..

The shallow copy may have other races, although I cannot think of any.

One alternative is to have another `debugChannel` along the `select {push, request}`. That way we serialize it; we can send on it a `[]string` of TypeUrls and then have a `debugResponseChannel` to send the responses back or something. Not sure if its better than this though, WDYT